xray/conventions.py

```diff
@@ -53,11 +55,16 @@ def mask_and_scale(array, fill_value=None, scale_factor=None, add_offset=None,
     """
     # by default, cast to float to ensure NaN is meaningful
     values = np.array(array, dtype=dtype, copy=True)
-    if fill_value is not None and not pd.isnull(fill_value):
-        if values.ndim > 0:
-            values[values == fill_value] = np.nan
-        elif values == fill_value:
-            values = np.array(np.nan)
+    if fill_value is not None and not np.all(pd.isnull(fill_value)):
+        if isinstance(fill_value, np.ndarray) and len(fill_value) != 1:
```

Comment:
This implicitly assumes that `fill_value` is 1D if it's an array.

I'm sure that's *probably* true, but it would be nice to have something slightly more robust, e.g., `if getattr(fill_value, 'size', 1) > 1:`